hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/EntityResultGraphNode.java

```diff
@@ -38,4 +40,14 @@ default JavaType<?> getResultJavaType() {
 	default EntityMappingType getReferencedMappingContainer() {
 		return getEntityValuedModelPart().getEntityMappingType();
 	}
+
+	@Override
+	default boolean appliesTo(GraphImplementor graphImplementor) {
+		assert graphImplementor.getGraphedType() instanceof EntityDomainType;
```

Comment:
This looks way better, but don't you think that this assert statement should rather be this?

```java
if ( !( graphImplementor.getGraphedType() instanceof EntityDomainType ) ) {
    return false;
}
```

I have no idea how this is called, so this could be totally wrong, just wanted to mention it so you give it a second thought.

Reply:
changed this method implementation, what do you think?